src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/BundlesRespondentController.java

```diff
@@ -0,0 +1,233 @@
+                @Content(mediaType = "application/json", schema = @Schema(implementation = CCDCallbackResponse.class))
+            }),
+        @ApiResponse(responseCode = "400", description = "Bad Request"),
+        @ApiResponse(responseCode = "500", description = "Internal Server Error")
+    })
+    public ResponseEntity<CCDCallbackResponse> aboutToSubmit(
+            @RequestBody CCDRequest ccdRequest,
+            @RequestHeader("Authorization") String userToken) {
+
+        boolean bundlesToggle = featureToggleService.isBundlesEnabled();
+        log.info(BUNDLES_LOG, bundlesToggle);
+        if (bundlesToggle) {
+            if (!verifyTokenService.verifyTokenSignature(userToken)) {
+                log.error(INVALID_TOKEN, userToken);
+                return ResponseEntity.status(FORBIDDEN.value()).build();
+            }
+
+            CaseData caseData = ccdRequest.getCaseDetails().getCaseData();
+            bundlesRespondentService.addToBundlesCollection(caseData);
+            bundlesRespondentService.clearInputData(caseData);
+            return getCallbackRespEntityNoErrors(caseData);
+        }
+        throw new ResponseStatusException(HttpStatus.SERVICE_UNAVAILABLE, BUNDLES_FEATURE_IS_NOT_AVAILABLE);
+    }
+
+    /**
+     * Populates the hearing list on page 3.
+     *
+     * @param ccdRequest holds the request and case data
+     * @param userToken  used for authorization
+     * @return Callback response entity with case data attached.
+     */
+    @PostMapping(value = "/midPopulateHearings", consumes = APPLICATION_JSON_VALUE)
+    @Operation(summary = "Populates the hearing list on page 3.")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "Accessed successfully",
+            content = {
+                @Content(mediaType = "application/json",
+                    schema = @Schema(implementation = CCDCallbackResponse.class))
+            }),
+        @ApiResponse(responseCode = "400", description = "Bad Request"),
+        @ApiResponse(responseCode = "500", description = "Internal Server Error")
+    })
+    public ResponseEntity<CCDCallbackResponse> midPopulateHearings(
+            @RequestBody CCDRequest ccdRequest,
+            @RequestHeader("Authorization") String userToken) {
+
+        boolean bundlesToggle = featureToggleService.isBundlesEnabled();
+        log.info(BUNDLES_LOG, bundlesToggle);
+        if (bundlesToggle) {
+            if (!verifyTokenService.verifyTokenSignature(userToken)) {
+                log.error(INVALID_TOKEN, userToken);
+                return ResponseEntity.status(FORBIDDEN.value()).build();
+            }
+            CaseData caseData = ccdRequest.getCaseDetails().getCaseData();
+            bundlesRespondentService.populateSelectHearings(caseData);
+            return getCallbackRespEntityNoErrors(caseData);
+        }
+        throw new ResponseStatusException(HttpStatus.SERVICE_UNAVAILABLE, BUNDLES_FEATURE_IS_NOT_AVAILABLE);
```

Comment:
minor: would prefer a guard clause here to reduce some nesting

if (!bundlesToggle) {
  throw .....
}
... rest of code

Reply:
Yeah I like that. Done.